lib/adobexd.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs');
+const path = require('path');
+const { promisify } = require('util');
+const unzip = require('unzipper');
+
+const { readFile, isDirectory } = require('./utils');
+
+function unzipXDFile(filePath, outputDir) {
+  return new Promise((resolve, reject) => {
+    fs.createReadStream(filePath, { emitClose: true })
+      .pipe(unzip.Extract({ path: outputDir }))
+      .on('error', reject)
+      .on('close', resolve)
+  });
+}
+
+async function removeUnzippedXDFile(outputDir) {
+  await promisify(fs.rmdir)(outputDir, { recursive: true });
+}
+
+async function getArtboardFiles(outputDir) {
+  const artboardSource = path.join(outputDir, 'artwork'); 
+  const directoryContents = await promisify(fs.readdir)(artboardSource);
+
+  const artboardDirs = directoryContents
+    .map(name => path.join(artboardSource, name))
+    .filter(isDirectory); // enkel desired contents houden
+  
+  const artboardFiles = await Promise.all(artboardDirs.map(directory => {
+    const fileLocation = path.join(directory, 'graphics', 'graphicContent.agc');
+    return readFile(fileLocation);
+  }));
+
+  // Only parse the first child, since we don't look at other children in an artboard
+  const artboardObjects = artboardFiles.map(artboard => JSON.parse(artboard).children[0]);
+
+  return artboardObjects;
+}
+
+function getReferencesFromManifest(manifest) {
+  const elementReferences = manifest.children[0].children;
+
+  // Only pick relevant object keys
+  // eslint-disable-next-line no-shadow
```

Comment:
why do you have to disable shadowing here?

Reply:
If this really is a problem, just return the mapped output

```
return elementReferences.map(…);
```

Reply:
Path is an object key we want to take, but path is also defined in the upper scope to join directory paths